src/Authentication/Authentication/Cmdlets/ConnectMgGraph.cs

```diff
@@ -196,20 +209,46 @@ private async Task ProcessRecordAsync()
                         break;
                 }
 
-                try
+                
+                var (context, authError) = await Authenticator.AuthenticateAsync(authContext, ForceRefresh, _cancellationTokenSource.Token,
+                   type =>
+                   {
+                       // Early Detection of FallBack to DeviceCode
+                       if (type == AuthProviderType.DeviceCodeProviderFallBack)
+                       {
+                           WriteWarning(Resources.DeviceCodeFallback);
+                       }
+                   });
+
+                switch (authError.AuthErrorType)
```

Comment:
Why not catch these in a try-catch? We are no longer relying on authError to fallback to device code.

Reply:
The question becomes for predictable scenarios, why throw an exception instead of handling the scenario?